src/middlewared/middlewared/alert/source/kmip.py

```diff
@@ -0,0 +1,15 @@
+from middlewared.alert.base import AlertClass, SimpleOneShotAlertClass, AlertCategory, AlertLevel
+
+
+class KMIPConnectionFailedAlertClass(AlertClass, SimpleOneShotAlertClass):
```

Comment:
You should mark this as `deleted_automatically = False` because you don't delete these alerts programmatically and instead they should be deleted when user dismisses them.